cudax/include/cuda/experimental/__event/event_ref.cuh

```diff
@@ -0,0 +1,166 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX_EVENT_REF_DETAIL_H
+#define _CUDAX_EVENT_REF_DETAIL_H
+
+/*
+    event_ref synopsis
+namespace cuda::experimental {
+class event_ref {
+public:
+    using value_type = cudaEvent_t;
+
+    event_ref(cudaEvent_t event_) noexcept : event(event_) {}
+
+    event_ref(int) = delete;
+    event_ref(nullptr_t) = delete;
+
+    void record(stream_ref) const;
+
+    void wait(stream_ref) const;
+
+    [[nodiscard]] cudaEvent_t get() const noexcept;
+
+    [[nodiscard]] explicit operator bool() const noexcept;
+
+    [[nodiscard]] friend bool operator==(event_ref, event_ref);
+    [[nodiscard]] friend bool operator!=(event_ref, event_ref);
+
+private:
+    cudaEvent_t event{}; // exposition only
+};
+}  // cuda::experimenal
+*/
+
+#include <cuda_runtime_api.h>
+// cuda_runtime_api needs to come first
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__cuda/api_wrapper.h>
+#include <cuda/std/__exception/cuda_error.h>
+#include <cuda/std/cassert>
+#include <cuda/std/cstddef>
+#include <cuda/std/utility>
+#include <cuda/stream_ref>
+
+namespace cuda::experimental
+{
+class event;
+class timed_event;
+
+//! @brief An non-owning wrapper for an untimed `cudaEvent_t`.
+class event_ref
+{
+private:
+  friend class event;
+  friend class timed_event;
+
+  ::cudaEvent_t __event_{0};
```

Comment:
AFAIK, `cudaEvent_t` is a [pointer](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html),

I would prefer if we either used
```suggestion
  ::cudaEvent_t __event_{};
```
or
```suggestion
  ::cudaEvent_t __event_{nullptr};
```